config/kibana.yml

```diff
@@ -6,11 +6,18 @@
 # To allow connections from remote users, set this parameter to a non-loopback address.
 #server.host: "localhost"
 
-# Enables you to specify a path to mount Kibana at if you are running behind a proxy. This only affects
-# the URLs generated by Kibana, your proxy is expected to remove the basePath value before forwarding requests
-# to Kibana. This setting cannot end in a slash.
+# Enables you to specify a path to mount Kibana at if you are running behind a proxy.
+# Use the `server.rewriteBasePath` setting to tell Kibana if it should remove the basePath
+# from requests it receives, and to prevent a deprecation warning at startup.
+# This setting cannot end in a slash.
 #server.basePath: ""
 
+# Specifies whether Kibana should rewrite requests that are prefixed with
+# `server.basePath` or require that they are rewritten by your reverse proxy.
+# This setting was effectively always `false` before Kibana 6.3 and will
+# default to `true` starting in Kibana 7.0.
```

Comment:
Do we have a way to document this for users? It's good that we have this in code comments, but it's probably good to make sure the change to defaults is documented in docs.